src/addons/phpmailer/lib/cronjob.php

```diff
@@ -0,0 +1,127 @@
+                            $logfilename = str_replace($dir."mail_log".DIRECTORY_SEPARATOR, "", $logfile);
+                        $zip->addFile($logfile, $logfilename);
+                        $zip->setEncryptionName($logfilename, ZipArchive::EM_AES_256, $this->getParam('password'));
+                    }
+                }
+            }
+    
+            $zip->close();
+
+            rex_dir::delete($dir.'mail_log', false);
+
+            $message = $file . $ext . ' created';
+
+            if ($this->getParam('delete_interval')) {
+                $allBackupFiles = glob(rex_path::addonData('phpmailer', '*.cronjob.zip'));
+                $backups = [];
+                $limit = strtotime('-1 month'); // Generelle Vorhaltezeit: 1 Monat
+
+                foreach ($allBackupFiles as $backupFile) {
+                    $timestamp = filectime($backupFile);
+
+                    if ($timestamp > $limit) {
+                        // wenn es die generelle Vorhaltezeit unterschreitet
+                        continue;
+                    }
+
+                    $backups[$backupFile] = $timestamp;
+                }
+
+                asort($backups, SORT_NUMERIC);
+
+                $countDeleted = 0;
+
+                foreach ($backups as $backup => $timestamp) {
+                    $stepLast = $step;
+                    $step = date($this->getParam('delete_interval'), (int) $timestamp);
+
+                    // dann löschen
+                    rex_file::delete($backup);
+                    ++$countDeleted;
+                }
+
+                if ($countDeleted) {
+                    $message .= ', '.$countDeleted.' old backup(s) deleted';
+                }
+            }
+
+            $this->setMessage($message);
+
+            return true;
+        }
+        $this->setMessage($file . $ext . ' not created');
+
+        return false;
+    }
+
+    public function getTypeName()
+    {
+        return rex_i18n::msg('phpmailer_logs');
+    }
+
+    public function getParamFields()
+    {
+        $fields = [
+            [
+                'label' => rex_i18n::msg('phpmailer_compressed_filename'),
+                'name' => 'filename',
+                'type' => 'text',
+                'default' => self::DEFAULT_FILENAME,
+                'notice' => rex_i18n::msg('phpmailer_compressed_filename_notice'),
+            ],
+             [
+                'label' => rex_i18n::msg('phpmailer_compressed_file_password'),
+                'name' => 'password',
+                'type' => 'text',
+             ],
+             [
+            'label' => rex_i18n::msg('phpmailer_delete_interval'),
+            'name' => 'delete_interval',
+            'type' => 'select',
+            'options' => [
+                '0' => rex_i18n::msg('phpmailer_delete_interval_off'),
+                'YW' => rex_i18n::msg('phpmailer_delete_interval_weekly'),
+                'YM' => rex_i18n::msg('phpmailer_delete_interval_monthly'), ],
```

Comment:
sollte man die flexibler halten?

Reply:
Alle x Tage?

Reply:
Logs werden in meinem Entwurf sofort gelöscht, wenn sie gezippt wurden. Das halte ich für sinnvoll. Die Häufigkeit bestimmt der Cronjob.

Es geht also tatsächlich um die ZIP-Dateien selbst. Was wäre da eine gute Auswahl?

Reply:
ich glaube wie skerbis sagt, alle X Tage ist am einfachsten und gibt dem User beliebig freiheiten.

Reply:
Ok, die Umsetzung schaffe ich hoffentlich bis Sonntag...

Reply:
Welchen Sonntag meintest du? :D

Im Ernst: Wäre klasse, wenn der PR auch in die 5.9 käme.
Gleichzeitig wäre ich aber froh, wenn entweder Alex oder jemand anderes ihn fertigstellt.

Reply:
> Logs werden in meinem Entwurf sofort gelöscht, wenn sie gezippt wurden. Das halte ich für sinnvoll.

Sofort gelöscht, wenn gezipt -> ja.
Aber frage mich, ob wirklich immer alle sofort gezipt werden sollten, wenn der Cronjob läuft, oder nicht nur welche älter x?

Reply:
Gut, dass ich mich nicht festgelegt habe :D

Ja, wäre ich auch froh. Ich habe erstmal gewartet, bis #3109 durch ist.

Reply:
> Aber frage mich, ob wirklich immer alle sofort gezipt werden sollten, wenn der Cronjob läuft, oder nicht nur welche älter x?

Ja, das wäre natürlich schöner, aber ich tue mich schwer darin, das überhaupt umzusetzen. Den PR brauche ich selbst nicht, da ich die Archive (formerly known as Logs) deaktiviere, es ist ein Geschenk an @tbaddade

Reply:
Habe gerade gesehen, dass wir den PR auch noch gar nicht in den Milestone gelegt haben.
Also falls niemand klar sagt, dass er den PR noch fertigstellt, lasse ich ihn auch draußen (und halt offen stehen).